Fix word break for repo pages

philipp-spiess opened the pull request:
Fixes #43687

Make sure that the header component allows long repo paths.

@AlicjaSuska: Path breadcrumbs also break very oddly in this view (they are displayed in the second line here?). What do you think of hiding some of the interim levels? e.g. if there are more than 8 directories or so, we hide some of the middle ones and just display a generic `...`? 

## Test plan

Tested locally:

<img width="1271" alt="Screenshot 2022-11-23 at 17 16 37" src="https://user-images.githubusercontent.com/458591/203792459-c9aae583-0b55-414d-95e9-6dbdc33f5eed.png">


<!-- All pull requests REQUIRE a test plan: https://docs.sourcegraph.com/dev/background-information/testing_principles -->

## App preview:

- [Web](https://sg-web-ps-fix-word-break-for-repo-pages.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

sg-e2e-regression-test-bob commented:
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.59 kb) | 0.01% (+1.59 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/fix-word-break-for-repo-pages/compare-report.html;tab=live_object) for a full comparison between the commits 64d14be and cc7bd73 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

AlicjaSuska reviewed (changes requested):
Thank you for making those changes @philipp-spiess. As we've discussed on slack:
- the main change of the title works well :) The only issue is the folder icon size - it is too small. Let's keep it the original size.
- regarding the breadcrumbs - I agree, this implementation is incorrect. Please, see the (old) figma [mockup](https://www.figma.com/file/A4gGoseJDz8iPeHP515MfQ/%F0%9F%A5%96Breaders-(breadcrumbs-%2B-headers)-%2315431-%5BApproved%5D?node-id=624%3A241&t=1O1clckqIfJOBwr4-4) for reference regarding the logic of adding '...' and which breadcrumb to show.

Thank you!

philipp-spiess commented:
@AlicjaSuska I'd like to work on the breadcrumbs changes in a different PR as they will quickly spiral out of control, do you mind reviewing this again for the page title only? I've also added an example to the storybook and updated the screenshot.
